devito/ir/support/space.py

```diff
@@ -3,6 +3,8 @@
 from functools import reduce
 from operator import mul
 
+from cached_property import cached_property
+
```

Comment:
nitpicking: no need for a blank line here.

The rule by which we organise imports is:
- first group are the imports from the Python library
- blank line
- second group are the important from third-party libraries
- blank line
- third group are the devito imports

Reply:
Thanks for the explanation. Space removed.

Assuming the new 'basic' profiling is enabled, in a basic problem on a 21x21 grid with 10,000 calls to op.apply() these changes being the time spent in `_prepare_arguments` from ~70% down to 40%.